mesa/agent.py

```diff
@@ -57,3 +60,41 @@ def advance(self) -> None:
     @property
     def random(self) -> Random:
         return self.model.random
+
+
+class AgentSet:
+    def __init__(self, agents: set[Agent], model: Model):
+        self._agents = agents
```

Comment:
Consider turning this into weak refs?

Reply:
Thanks for implementing, curious how it looks, will review the implementation tomorrow.